### PR TITLE
Add Okta SAML guide

### PR DESCRIPTION
**Pull request by sixhobbits**

### CODE REVIEW

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+:sectnumlevels: 0
+
+== Configure SAML v2 for Okta
+
+This documentation will guide you in configuring SAML v2 IdP for Okta. In this case, FusionAuth will act as Service Provider (SP) to Okta (IdP). Functionally, the result will allow you to display a [uielement]#Login with Okta# button on your FusionAuth application's login page and connect via SAML to Okta users/applications.
+
+image:identity-providers/okta-saml/okta-saml-result.png[result-intro,width=1200,role=bottom-cropped]
+
+== Prerequisites
+* A FusionAuth instance running on a publicly accessible URL. You can spin up a link:/pricing[basic FusionAuth Cloud instance] or link:/docs/v1/tech/installation-guide/[install it on any server].
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:17]:
```suggestion
* A FusionAuth instance running on a publicly accessible URL. You can spin up a link:/pricing[basic FusionAuth Cloud instance], use link:/docs/v1/tech/developer-guide/exposing-instance[ngrok], or link:/docs/v1/tech/installation-guide/[install it on any server].
```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+* An link:https://www.okta.com/free-trial/[Okta Workforce Identity Cloud account] connected to a business email address.
+** The link:https://help.okta.com/eu/en-us/Content/Topics/end-user/ov-overview.htm?cshid=csh-user-ov-overview[Okta Verify app] if your Okta account is not configured to send SMS messages for 2FA.
+
+== Okta Configuration
+
+Log in to Okta and navigate to the [uielement]#Admin# panel.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:23]:
```suggestion
Log in to Okta and navigate to the [breadcrumb]#Admin# panel.
```

If you are navigating someplace, use breadcrumb. If it is a field, use field. If it is in the UI but anything else, use uielement.

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+image::identity-providers/okta-saml/okta-saml-assign-to-people.png[assign to people,width=1200,role=bottom-cropped]
+
+Click the [uielement]#Assign# button next to the user you would like to authenticate. Click [uielement]#Save and Go Back#, then [uielement]#Done#.
+
+image::identity-providers/okta-saml/okta-saml-assign-button.png[assign button,width=1200]
+
+== FusionAuth Configuration
+
+From your FusionAuth dashboard, navigate to [breadcrumb]#Settings -> Key Master# and click the downward arrow next to [uielement]#Generate Elliptic#. Then click [uielement]#Import Certificate#.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:61]:
```suggestion
Open a new tab and go to your FusionAuth dashboard.

From your FusionAuth dashboard, navigate to [breadcrumb]#Settings -> Key Master# and click the downward arrow next to [uielement]#Generate Elliptic#. Then click [uielement]#Import Certificate#.
```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+Navigate to [breadcrumb]#Settings -> Identity Providers# and add a new `SAML v2` identity provider.
+
+image::identity-providers/okta-saml/okta-saml-new-idp.png[new identity provider,width=1200,role=bottom-cropped]
+
+Provide a [field]#Name#, for example, `OKTA SAML`. For the [field]#IdP endpoint#, paste the [field]#Identity Provider Single Sign-On URL# from the Okta [breadcrumb]#SAML setup instructions# page. For the verification key, select from the dropdown the name of the key you just created in the Key Master. Optionally, change the [field]#Button text# to `Login with Okta`.
+
+image::identity-providers/okta-saml/okta-saml-new-idp-settings.png[new idp settings,width=1200]
+
+Make sure your application has the [uielement]#Enabled# switch toggled. Then, navigate to the [breadcrumb]#Options# tab and supply the names that you chose for the [field]#user.Id# and [field]#user.email# attribute statements in Okta.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:77]:
```suggestion
Make sure your application has the [field]#Enabled# switch toggled. Then, navigate to the [breadcrumb]#Options# tab and supply the names that you chose for the [field]#user.Id# and [field]#user.email# attribute statements in Okta.
```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+image::identity-providers/okta-saml/okta-saml-claim-options.png[claim options,width=1200]
+
+Hit [uielement]#Save# and [uielement]#View# the identity provider you just created. Scroll down to the [breadcrumb]#SAML v2 Integration details# section and record these values so you can supply them to Okta.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:81]:
```suggestion
Hit [uielement]#Save#. Then [uielement]#view# the identity provider you just created (using the green magnifying glass). Scroll down to the [breadcrumb]#SAML v2 Integration details# section and record these values so you can supply them to Okta.
```

Comment (mooreds) [reply site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:81]: Which values do you need to record? All of them? Please list them out here.

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+
+image::identity-providers/okta-saml/okta-saml-cors-warning.png[cors warning,width=1200,role=bottom-cropped]
+
+image::identity-providers/okta-saml/okta-saml-cors-settings.png[cors settings,width=1200]
+
+Return to your Okta dashboard. Navigate to [breadcrumb]#Applications -> Your Application# and edit the [breadcrumb]#SAML Settings#.
+
+image::identity-providers/okta-saml/okta-saml-edit-saml-settings.png[edit SAML settings,width=1200,role=bottom-cropped]
+
+Click [uielement]#Next# to get to the [breadcrumb]#Configure SAML# tab. Replace the [field]#Single sign-on URL# with the [field]#Callback URL (ACS)# integration detail from FusionAuth. Replace the [field]#Audience URI (SP Entity ID)# with the [field]#Issuer# integration detail from FusionAuth. Click [uielement]#Next# and [uielement]#Finish# to save these settings.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:95]:
```suggestion
Click [uielement]#Next# to get to the [breadcrumb]#Configure SAML# tab. Replace the [field]#Single sign-on URL# with the [field]#Callback URL (ACS)# value from FusionAuth. Replace the [field]#Audience URI (SP Entity ID)# with the [field]#Issuer# value from FusionAuth. Click [uielement]#Next# and then [uielement]#Finish# to save these settings.
```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+Make sure your application has the [uielement]#Enabled# switch toggled. Then, navigate to the [breadcrumb]#Options# tab and supply the names that you chose for the [field]#user.Id# and [field]#user.email# attribute statements in Okta.
+
+image::identity-providers/okta-saml/okta-saml-claim-options.png[claim options,width=1200]
+
+Hit [uielement]#Save# and [uielement]#View# the identity provider you just created. Scroll down to the [breadcrumb]#SAML v2 Integration details# section and record these values so you can supply them to Okta.
+
+image::identity-providers/okta-saml/okta-saml-integration-details.png[integration details,width=1200]
+
+If you receive a warning after saving the identity provider, navigate to [breadcrumb]#Settings -> System# and add your Okta URL to the [field]#Allowed origins# list.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:85]:
```suggestion
You may receive a warning after saving the identity provider.
```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+image::identity-providers/okta-saml/okta-saml-cors-warning.png[cors warning,width=1200,role=bottom-cropped]
+
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:88]:
```suggestion
If you receive such a warning after saving the identity provider, navigate to [breadcrumb]#Settings -> System# and add your Okta URL to the [field]#Allowed origins# list. This configures FusionAuth to send the correct CORS headers.

```

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+== Configure SAML v2 for Okta
+
+This documentation will guide you in configuring SAML v2 IdP for Okta. In this case, FusionAuth will act as Service Provider (SP) to Okta (IdP). Functionally, the result will allow you to display a [uielement]#Login with Okta# button on your FusionAuth application's login page and connect via SAML to Okta users/applications.
+
+image:identity-providers/okta-saml/okta-saml-result.png[result-intro,width=1200,role=bottom-cropped]
+
+== Prerequisites
+* A FusionAuth instance running on a publicly accessible URL. You can spin up a link:/pricing[basic FusionAuth Cloud instance] or link:/docs/v1/tech/installation-guide/[install it on any server].
+* An link:https://www.okta.com/free-trial/[Okta Workforce Identity Cloud account] connected to a business email address.
+** The link:https://help.okta.com/eu/en-us/Content/Topics/end-user/ov-overview.htm?cshid=csh-user-ov-overview[Okta Verify app] if your Okta account is not configured to send SMS messages for 2FA.
```
[the remark this answers is not in the document]
Comment (mooreds) [reply site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:19]: Why do we need 2fa here?

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+image::identity-providers/okta-saml/okta-saml-settings-from-fusionauth.png[settings from fusionauth,width=1200]
+
+== Testing the Login
+
+To test, navigate to the applications page in FusionAuth. Click on the [uielement]#View# icon (magnifying glass) next to the application you enabled Okta login on and copy the [field]#OAuth IdP login URL# address. Navigate to this address. You should see a [uielement]#Login with Okta# option on your app's sign-in page:
+
+image::identity-providers/okta-saml/okta-saml-result.png[result,width=1200,role=bottom-cropped]
+
+Click the [uielement]#Login with Okta# button. Test logging in with the username and password, as well as the 2FA method you selected when you configured your account on Okta.
```
Comment (mooreds) [inline site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:105]: Not sure why 2fa is set up for okta? How does that help the tutorial?

**Review by mooreds — CHANGES_REQUESTED** [review]
Convert we to you--saw `we` used at least once.

it's CORS not cors in the alt text

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc (hunk 10 above)
Comment (TheMiniDriver) [reply site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc:105]: Okta enforces using a form of 2FA when creating a new account, so we thought it best to mention it in the guide. 

Screenshot 2023-03-09 at 16 22 51

File: site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc (hunk 5 above)
Comment (TheMiniDriver) [reply site/docs/v1/tech/identity-providers/samlv2/okta-saml.adoc]: This has been updated to read: 
"Hit [uielement]#Save# and [uielement]#View# the identity provider you just created. Scroll down to the [breadcrumb]#SAML v2 Integration details# section and record the [field]#Callback URL (ACS)# and [field]#Issuer# values so you can supply them to Okta."

**Comment by TheMiniDriver** [comment]
Hi @mooreds, we've applied the changes you've suggested, and made a note of them for future guides. Thanks!

**Review by mooreds — APPROVED** [review]
Looks great, thanks!

**Pull request merged**